core/os/memory.cpp

```diff
@@ -65,6 +66,38 @@ SafeNumeric<uint64_t> Memory::max_usage;
 
 SafeNumeric<uint64_t> Memory::alloc_count;
 
+inline bool is_power_of_2(size_t x) { return x && ((x & (x - 1U)) == 0U); }
+
+void *Memory::alloc_aligned_static(size_t p_bytes, size_t p_alignment) {
+	DEV_ASSERT(is_power_of_2(p_alignment));
+
+	void *p1, *p2;
+	if ((p1 = (void *)malloc(p_bytes + p_alignment - 1 + sizeof(uint32_t))) == nullptr) {
+		return nullptr;
+	}
+
+	p2 = (void *)(((uintptr_t)p1 + sizeof(uint32_t) + p_alignment - 1) & ~((p_alignment)-1));
+	*((uint32_t *)p2 - 1) = (uint32_t)((uintptr_t)p2 - (uintptr_t)p1);
+	return p2;
+}
+
+void *Memory::realloc_aligned_static(void *p_memory, size_t p_bytes, size_t p_prev_bytes, size_t p_alignment) {
+	if (p_memory == nullptr) {
+		return alloc_aligned_static(p_bytes, p_alignment);
+	}
+
+	void *ret = alloc_aligned_static(p_bytes, p_alignment);
+	memcpy(ret, p_memory, p_prev_bytes);
+	free_aligned_static(p_memory);
+	return ret;
+}
```

Comment:
I'd say this has to be on `realloc_static()`, to avoid pessimizing for the case where the address changes.

Reply:
Could you elaborate? Because I do not understand what you are asking.

Reply:
My point is that a reallocation doesn't necessarilly involve a new address. When the allocator can just resize the block and give you the same address, you can just avoid any further operations. In contrast, this implementation will never benefit from that and will unconditionally pay for the overhead of dealing with a second allocation and, more important, that of copying the data.

Reply:
aligned_realloc is a minefield and relatively [hard to implement](https://stackoverflow.com/questions/56619623/why-is-there-no-aligned-realloc-on-most-platforms).

Not impossible, but not straightforward (it looks straightforward, then the devil in the details appear). I would advise against it unless someone wants to write heavy unit testing to ensure it works in all edge cases.

Reply:
Let's go with this one. I'll submit a separate PR with a more efficient implementation if I can come up with any.